src/main/java/com/commercetools/sync/categories/utils/CategoryAssetUpdateActionUtils.java

```diff
@@ -34,9 +35,12 @@
      * @return A list with the update actions or an empty list if the asset fields are identical.
      */
     @Nonnull
-    public static List<UpdateAction<Category>> buildActions(@Nonnull final Asset oldAsset,
-                                                            @Nonnull final AssetDraft newAsset,
-                                                            @Nonnull final CategorySyncOptions syncOptions) {
+    public static  <D> List<UpdateAction<Category>> buildActions(
+        @Nonnull final Resource oldRessource,
+        @Nonnull final D newRessource,
```

Comment:
Why is `newResource` of type `D`? What does it mean? Why not `Object`? What is the benefit of giving it that `D`?

Reply:
723fc494